Feature/cpesp 5462 aidl versioning for rdk e hal new

aniketp1809 opened the pull request.

aniketp1809 committed:
CPESP-5462: AIDL Versioning for RDK-E HAL

Initial Changes for the AIDL Versioning
1. CMake configuration
2. Implementation of versioning
3. Example aidls for validations

Reviewed by: Varghese Mathai, Rahul Raveendran
JIRA URL: https://ccp.sys.comcast.net/browse/CPESP-5462

aniketp1809 committed:
CPESP-5462: AIDL Versioning for RDK-E HAL

1. Modified CMake Configuration to support Yocto Builds
2. Updated Scripts to locate bash scripts correctly.
3. Updated Python scripts to work with python 3.8.10

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 0 files pending approval, 1 file pending identification.
* Protex Server Path: /home/blackduck/github/linux_binder_idl/7/rdkcentral/linux_binder_idl

* Commit: 293685517fbade3ec7a549e10e9d097bda0c248e

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/ce7693f751d43d4e1db3b5e5c7c11e71)'

rdkcmf-jenkins commented:
b'## **WARNING:** A Blackduck scan failure has been waived

A prior failure has been upvoted
* Upvote reason: OK

* Commit: 293685517fbade3ec7a549e10e9d097bda0c248e
'

mhughesacn commented:
I am sorry - should have checked PR#6 yesterday.

aniketp1809 committed:
CPESP-5462: AIDL Versioning for RDK-E HAL

Code Clean up and added required comments for function
definitions

Reviewed by: Varghese Mathai, Rahul Raveendran
JIRA URL: https://ccp.sys.comcast.net/browse/CPESP-5462

aniketp1809 committed:
CPESP-5462: AIDL Versioning for RDK-E HAL

Adapted scripts for new directory layout.
Better variable names for the readability
Updated CMake Configuration to handle compilation of stubs and proxies.
Added headers for licensing
Installing aidl_ops in INSTALL_DIR/bin
Do Not Generate sources again for frozen versions.
Added ENV variable to skip specific directories while checking for interfaces

Reviewed by: Varghese Mathai, Rahul Raveendran
JIRA URL: https://ccp.sys.comcast.net/browse/CPESP-5462

aniketp1809 commented:
Hi @mhughesacn 
Could you please check in this repo if all the licensing-related information is in place?
Thank You in Advance,
Regards,
Aniket

mhughesacn commented:
Hi @aniketp1809 : Happy with this, thank you.

rahul-pr-87 reviewed (approved):
Looks good
